src/target/riscv/riscv.c

```diff
@@ -5847,6 +5848,52 @@ bool reg_is_initialized(const struct reg *reg)
 	return true;
 }
 
+static struct target *reg_get_target(const struct reg *reg)
+{
+	assert(reg_is_initialized(reg));
+	return ((const riscv_reg_info_t *)reg->arch_info)->target;
+}
+
+static int riscv011_reg_get(struct reg *reg)
+{
+	struct target * const target = reg_get_target(reg);
+	RISCV_INFO(info);
+	riscv_reg_t value;
+	const int result = info->get_register(target, &value, reg->number);
+	if (result != ERROR_OK)
+		return result;
+	buf_set_u64(reg->value, 0, reg->size, value);
+	return ERROR_OK;
+}
+
+static int riscv011_reg_set(struct reg *reg, uint8_t *buf)
+{
+	const riscv_reg_t value = buf_get_u64(buf, 0, reg->size);
+	struct target * const target = reg_get_target(reg);
+	RISCV_INFO(info);
+	return info->set_register(target, reg->number, value);
+}
+
+static struct reg_arch_type *gdb_regno_reg_type(const struct target *target,
+		uint32_t regno)
+{
+	RISCV_INFO(info);
+	assert(info);
+	if (info->dtm_version == DTM_DTMCS_VERSION_0_11) {
+		static struct reg_arch_type riscv011_type = {
+			.get = riscv011_reg_get,
+			.set = riscv011_reg_set
+		};
+		return &riscv011_type;
+	}
+
+	static struct reg_arch_type default_type = {
+		.get = register_get,
+		.set = register_set
+	};
+	return &default_type;
+}
```

Comment:
1) I am not a fan of static items inside functions since this way the lifetime of the object does not match the scope (in another words, these are hidden global variables). I'd prefer to define the static structures on the global scope - if you'd like.

2) Please, rename the below items for clarity and consistency:

- `riscv011_type` --> `riscv011_reg_type`
- `default_type` --> `riscv013_reg_type`
- `register_get` --> `riscv013_reg_get`
- `register_set` --> `riscv013_reg_set`

Reply:
1. I prefer to limit the scope of variables as much as possible, so I would leave it as-is, if it's fine by you.
2. Addressed.

Reply:
Ok, let's leave the current state.

In future, when `riscv.c` is split into multiple smaller modules, the scope of the static variables will be smaller due to smaller modules, so this will become less of a concern.